### PR TITLE
Contact details required for data map and ropa

### LINKED ISSUE #321: Resource Updates for Controller Contact Details
### Is your feature request related to a specific problem?

As part of exporting a data map (or eventually a ROPA) from `fidesctl`, additional information around the organization is required compared to what is maintained today. This issue will focus specifically on [section 1.a of article 30](https://gdpr-info.eu/art-30-gdpr/) and can be worked concurrently with #322 & #323

### Describe the solution you'd like

The following items will need to be added as part of the walk state of this effort to get us to having an exportable data map as a csv:

* Contact Information for the controller, data protection officer, and a representative, likely managed as part of the organization resource
    * Name
    * Address
    * Phone
    * Email
* Joint controller and processor details
    * In the example this is listed as a single field tied to a row (or rows) in the ROPA - may need some extra brain power around displaying this
    * Feels like it could be managed at the collection level if understanding correctly (I.e. Zendesk may be a joint controller?)

### Describe alternatives you've considered, if any

N/A

### Additional context

This is part of a larger group of issues that will continue to update resource models to add attributes and data points that will make up the data map and eventual ROPA. Minus the Joint controller and processor details, this issue is the clearest in terms of where the data should be applied. If more discussion is required, the joint controller and processor details can be offloaded to a later issue in the spirit of momentum

### PR DESCRIPTION
**Pull request by SteveDMurphy**
Closes #321

### Code Changes

* [x] Add a `ContactDetail` class
* [x] Add fields populated with `ContactDetail` to organization, system, and dataset resources
* [x] Create `demo_organization.yml`
* [x] Identify all places where documentation updates are required or would be helpful
* [x] ~Look at creating an export command here (potentially) to ensure data usage makes sense~ discussing a different tactic for this

### Steps to Confirm

* [x] Validated that the new data is correctly applied to the server

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* [x] Documentation Updated

### Description Of Changes

This PR is one of many steps towards creating a valid and useful data map exported from fidesctl. The fields added are all related to required and optional contact information. A new class for standardized contact details was created for this and applied to organizations, systems, and datasets.

Although the contact information should technically be public facing information, the values are encrypted at rest on the database using the `pgcrypto` extension within postgres.

### CODE REVIEW

**Comment by ThomasLaPiana** [comment]
@SteveDMurphy taking a look at this later today, thank you!

File: fidesctl/src/fideslang/models.py [hunk 1]
```diff
@@ -117,6 +117,23 @@ class DatasetCollection(BaseModel):
...
+    The contact details information model.
+
+    Used to capture contact information for controllers, used
+    as part of exporting a data map / ROPA.
+
+    This model is nested under an Organization and
+    potentially under a system/dataset.
+    """
+
+    name: str = ""
```
[the remark this answers is not in the document]
Comment (ThomasLaPiana) [reply fidesctl/src/fideslang/models.py:131]: We should be* encrypting this

**Review by ThomasLaPiana — CHANGES_REQUESTED** [review]
Overall looks great, but need to be encrypting the contact info in the database

File: fidesctl/src/fideslang/models.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (SteveDMurphy) [reply fidesctl/src/fideslang/models.py:131]: Oof, thank you! Can't believe I spaced on adding that 🤦🏽

[the remark this answers is not in the document]
Comment (SteveDMurphy) [reply fidesctl/src/fideslang/models.py:131]: I was finally able to work through the remaining issues I was having, although after looking at `fidesops` it seems like there may be a better way of doing it (or at minimum just stay consistent with them) - let me know what you think @ThomasLaPiana , thanks for the help in reviewing this!

File: docs/fides/docs/language/resources/organization.md [hunk 2]
```diff
@@ -31,13 +41,34 @@ organization:
...
-  "description": "An Organization that represents all of Acme Inc."
+  "description": "An Organization that represents all of Acme Inc.",
+  "controller": {
+    "name": "Dave L. Epper",
+    "address": "1 Acme Pl. New York, NY",
+    "email": "controller@acmeinc.com",
+    "phone": "+1 555 555 5555"
+  },
+  "data_protection_officer": {
+    ...
```
[the remark this answers is not in the document]
Comment (ThomasLaPiana) [reply docs/fides/docs/language/resources/organization.md:68]: i would prefer to make these fully valid yaml, as we may want to start automatically validating these are part of our docs build process

File: fidesctl/src/fideslang/models.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ThomasLaPiana) [reply fidesctl/src/fideslang/models.py:131]: this option seems fine to me! we'll eventually get crypto added to fideslib and can pull from there, but I'm happy with this implementation

**Comment by ThomasLaPiana** [comment]
@SteveDMurphy last thing i can see is we need to update the fides resources! merge main to get up-to-date with those and then go ahead and add the new columns and categories!

**Review by ThomasLaPiana — CHANGES_REQUESTED** [review]
LGTM! Just need to update the fides resources for the project

**Comment by SteveDMurphy** [comment]
> LGTM! Just need to update the fides resources for the project

Great catch @ThomasLaPiana! I _think_ I added everything correctly (for now, more to come over the next few issues) and ran the evaluation, but please let me know if anything looks a little wonky. Thanks again!

**Review by ThomasLaPiana — APPROVED** [review]
LGTM!

**Pull request merged**